packages/report-generator-runner/src/runner/runner.ts

```diff
@@ -3,152 +3,125 @@
-            return;
-        }
-
-        this.telemetryManager.trackScanStarted(scanMetadata.scanGroupId);
+        this.telemetryManager.trackRequestStarted(runMetadata.scanGroupId);
         try {
-            const privacyScanResults = {}; // TBD
-            await this.processScanResult(privacyScanResults, pageScanResult);
+            const queuedRequests = await this.requestSelector.getQueuedRequests(this.maxQueuedRequests);
+            await this.updateRequestStateToRunning(queuedRequests);
+            this.reportProcessor.generate(queuedRequests.requestsToProcess);
+            this.selectCompletedRequestsToDelete(queuedRequests);
+            await this.updateRequestStateToFailed(queuedRequests.requestsToProcess);
+            await this.deleteRequests(queuedRequests.requestsToDelete);
+            await this.updateScanRunStatesToCompleted(queuedRequests.requestsToDelete);
         } catch (error) {
-            const errorMessage = System.serializeError(error);
-            this.setRunResult(pageScanResult, 'failed', errorMessage);
-
-            this.logger.logError(`The report generator processor failed to scan a webpage.`, { error: errorMessage });
-            this.telemetryManager.trackScanTaskFailed();
+            this.logger.logError(`The report generator processor failed.`, { error: System.serializeError(error) });
+            this.telemetryManager.trackRequestFailed();
         } finally {
-            this.telemetryManager.trackScanCompleted();
+            this.telemetryManager.trackRequestCompleted();
+            this.logger.logInfo('Stop report generator runner.');
         }
-
-        await this.updateScanResult(scanMetadata, pageScanResult);
-
-        this.logger.logInfo('Stop report generator runner.');
     }
 
-    private async updateScanRunState(scanId: string): Promise<OnDemandPageScanResult> {
-        this.logger.logInfo(`Updating webpage scan run state to 'running'.`);
-        const partialPageScanResult: Partial<OnDemandPageScanResult> = {
-            id: scanId,
-            run: {
-                state: 'running',
-                timestamp: new Date().toJSON(),
-                error: null,
-            },
-            scanResult: null,
-            reports: null,
-        };
-        const response = await this.onDemandPageScanRunResultProvider.tryUpdateScanRun(partialPageScanResult);
-        if (!response.succeeded) {
-            this.logger.logWarn(
-                `Update webpage scan run state to 'running' failed due to merge conflict with other process. Exiting webpage scan task.`,
-            );
-
-            return undefined;
-        }
-
-        return response.result;
+    private selectCompletedRequestsToDelete(queuedRequests: QueuedRequests): void {
+        const completedRequests = queuedRequests.requestsToProcess.filter((queuedRequest) => queuedRequest.condition === 'completed');
+        queuedRequests.requestsToDelete.push(...completedRequests);
```

Comment:
Should completed requests also be removed from requestsToProcess so they aren't marked as failed?

Reply:
Updated